manifest-editor/src/hooks/useAnnotationsList.tsx

```diff
@@ -23,60 +25,85 @@ export function getInitialAnnotationList(canvasId: string) {
   return annos;
 }
 
+export function getAnnotationsByAnnotationPage(annotationPageID?: string) {
+  const vault = useVault();
+
+  const annotationPage = useAnnotationPage({ id: annotationPageID });
+  const annos: any[] = [];
+  if (annotationPage) {
+    annotationPage.items.map((item: any) => {
+      annos.push(vault.get(item.id));
+    });
+  }
+  return annos;
+}
+
 export function useAnnotationList<T = AnnotationNormalized>(
-  canvasId: string
+  canvasId: string,
+  annotationPageId?: string
 ): AnnotationNormalized[] | T | undefined | [] | any {
   const [isEditing, setIsEditing] = useState<boolean>(false);
   const [selectedAnnotation, setSelectedAnnotation] = useState<string>();
   const [annotations, setAnnotations] = useState<any[]>(getInitialAnnotationList(canvasId));
   const manifest = useManifest();
   const vault = useVault();
+  const [pagedAnnotations, setPagedAnnotations] = useState<any[]>(getAnnotationsByAnnotationPage(annotationPageId));
+  const annotationPage = useAnnotationPage();
 
   // @todo this will need to know which annotation page is being referenced
-  const addNewAnnotation = useCallback((bounds: { x: number; y: number; width: number; height: number }) => {
-    const id = v4();
-    // we need more than these details here
-    setAnnotations((a) => [...a, { id, ...bounds }]);
-    setIsEditing(false);
-    setSelectedAnnotation(undefined);
+  const addNewAnnotation = useCallback(
+    (
+      bounds: { x: number; y: number; width: number; height: number } = {
+        x: 0,
+        y: 0,
+        width: 100,
+        height: 200,
+      }
+    ) => {
+      const id = `https://example.org/annotation/${v4()}`;
+      // we need more than these details here
+      const target = `${canvasId}#xywh=${bounds.x},${bounds.y},${bounds.width},${bounds.height}`;
+      setAnnotations((a) => [...a, { id, target }]);
+      setIsEditing(false);
+      setSelectedAnnotation(undefined);
+      console.log(annotationPage);
 
-    // @todo get this working
-    // const builder = new IIIFBuilder(vault);
-    // builder.editManifest(manifest.id, (mani: any) => {
-    //   mani.editCanvas(canvas.id, (can: any) => {
-    //     can.createAnnotation(canvas.id, {
-    //       id: `${newID}/annotation-page`,
-    //       type: "AnnotationPage",
-    //       motivation: "describing",
-    //       body: {
-    //         id: v4(),
-    //         type: "TextualBody",
-    //         format: "text/html",
-    //         height: 500,
-    //         width: 500,
-    //         target: "string",
-    //       },
-    //     });
-    //   });
-    // });
-  }, []);
+      vault.dispatch(
+        importEntities({
+          entities: {
+            [`${id}/annotation`]: {
+              id: `${id}/annotation`,
```

Comment:
there is an import for `emptyAnnotation` and `emptyAnnotationPage` to get the other empty fields. Usually I just spread them:
```ts
[`${id}/annotation`]: {
    ...emptyAnnotation,
    id: `${id}/annotation`,
}
```

Reply:
@stephenwf where does the import come from?